lib/CAPI/Dialect/OM.cpp

```diff
@@ -140,6 +140,17 @@ MlirType omEvaluatorObjectGetType(OMEvaluatorValue object) {
   return wrap(llvm::cast<Object>(unwrap(object).get())->getType());
 }
 
+/// Get the hash for the object.
+unsigned omEvaluatorObjectGetHash(OMEvaluatorValue object) {
+  return llvm::cast<Object>(unwrap(object).get())->getHash();
```

Comment:
Rather than defining the getHash methods on each EvaluatorValue, would it be sufficient to just do `llvm::hash_value(unwrap(object).get())`? I.e., just consider the objects' pointers for unique hashes? I think we take care to allocate unique objects in the evaluator. What do you think @uenoku?

Reply:
Yes, I think `llvm::hash_value(unwrap(object).get())` is sufficient. Evaluator should take care of allocations.